docs/markdown/Tutorials/test-custom-plugin-goal.md

```diff
@@ -0,0 +1,280 @@
+# Introduction
+
+In this tutorial, we'll learn how to test the custom plugin we wrote earlier. Pants documentation provides comprehensive coverage of the [plugin testing](https://www.pantsbuild.org/docs/rules-api-testing) and this tutorial should you to get started writing own tests.
+
+Most of the plugin code that needs to be tested is in the following files:
+* `rules.py` where we implemented how a `VERSION` file needs to be read and how to use a `version_file` BUILD target
+* `tailor.py` where we taught the `tailor` goal about the `VERSION` files and generation of `version_file` targets
+
+To get started writing tests, it may make sense to write a very high level integration test first to confirm our code does what we expect. You can think of this as of [integration tests for Pants](https://www.pantsbuild.org/docs/rules-api-testing#approach-4-run_pants-integration-tests-for-pants). Let's run our goal from a test!
```

Comment:
"think of this as of .."  perhaps grammatically correct, but hard to follow the rythm (for me) :)

Reply:
Made it simpler!